src/request.rs

```diff
@@ -898,12 +937,12 @@ impl<'a> Request<'a> {
         stream.set_write_timeout(self.write_timeout)?;
 
         if self.inner.uri.scheme() == "https" {
-            let mut cnf = tls::Config::default();
-            let cnf = match self.root_cert_file_pem {
-                Some(p) => cnf.add_root_cert_file_pem(p)?,
-                None => &mut cnf,
-            };
-            let mut stream = cnf.connect(host, stream)?;
+            let mut stream = config
+                .ok_or_else(|| {
+                    println!("No tls config defined for the https request!");
```

Comment:
Also here no `println!` if possible. If it makes sense, add this information to the error, or log it.